planemo/commands/cmd_test.py

```diff
@@ -73,23 +77,32 @@ def cli(ctx, paths, **kwds):
     against that same Galaxy root - but this may not be bullet proof yet so
     please careful and do not try this against production Galaxy instances.
     """
-    runnables = for_paths(paths)
-    is_cwl = all([r.type in [RunnableType.cwl_tool, RunnableType.cwl_workflow] for r in runnables])
-    if kwds.get("engine", None) is None:
-        kwds["engine"] = "galaxy" if not is_cwl else "cwltool"
+    with temp_directory(dir=ctx.planemo_directory) as temp_path:
+        # Create temp dir(s) outside of temp, docker can't mount $TEMPDIR on OSX
+        runnables = for_paths(paths, temp_path=temp_path)
+        is_cwl = all([r.type in [RunnableType.cwl_tool, RunnableType.cwl_workflow] for r in runnables])
+        if kwds.get("engine", None) is None:
+            kwds["engine"] = "galaxy" if not is_cwl else "cwltool"
 
-    engine_type = kwds["engine"]
-    enable_test_engines = any([r.type not in [RunnableType.galaxy_tool, RunnableType.galaxy_datamanager, RunnableType.directory] for r in runnables])
-    enable_test_engines = enable_test_engines or engine_type != "galaxy"
-    if enable_test_engines:
-        ctx.vlog("Using test engine type %s" % engine_type)
-        with engine_context(ctx, **kwds) as engine:
-            test_data = engine.test(runnables)
-            return_value = handle_reports_and_summary(ctx, test_data.structured_data, kwds=kwds)
-    else:
-        ctx.vlog("Running traditional Galaxy tool tests using run_tests.sh in Galaxy root %s" % engine_type)
-        kwds["for_tests"] = True
-        with galaxy_config(ctx, runnables, **kwds) as config:
-            return_value = run_in_config(ctx, config, **kwds)
+        engine_type = kwds["engine"]
+        test_engine_testable = (RunnableType.galaxy_tool, RunnableType.galaxy_datamanager, RunnableType.directory)
+        enable_test_engines = any(r.type not in test_engine_testable for r in runnables)
+        enable_test_engines = enable_test_engines or engine_type != "galaxy"
+        if enable_test_engines:
+            ctx.vlog("Using test engine type %s" % engine_type)
+            with engine_context(ctx, **kwds) as engine:
+                test_data = engine.test(runnables)
+                return_value = handle_reports_and_summary(ctx, test_data.structured_data, kwds=kwds)
+        else:
+            ctx.vlog("Running traditional Galaxy tool tests using run_tests.sh in Galaxy root %s" % engine_type)
+            kwds["for_tests"] = True
+            with galaxy_config(ctx, runnables, **kwds) as config:
+                return_value = run_in_config(ctx, config, **kwds)
+                if kwds.get('update_test_data'):
+                    non_copied_runnables = for_paths(paths)
+                    target = _find_test_data(non_copied_runnables, **kwds)
+                    if config.test_data_dir != target:
+                        ctx.vlog("Copying %s to %s" % (config.test_data_dir, target))
+                        copy_tree(config.test_data_dir, target)
```

Comment:
Shouldn't lines 99-106 be inside `run_in_config()` ?

Reply:
I guess we could, but we only want to do this for planemo test, which is why I placed it here (If we do this for planemo serve edits to the tool files won't be reloaded)

Reply:
`planemo serve` does not call `run_in_config()`, only `test` and `shed_test` do.

The reason to put it there would be that it's where the normal test data update is executed: https://github.com/galaxyproject/planemo/blob/master/planemo/galaxy/test/actions.py#L98-L100

Reply:
That also gives us the chance to only copy the files once. Le't see how https://github.com/galaxyproject/planemo/pull/988/commits/8656796d9fd9ae3900a84bc80e2894137499365f fares